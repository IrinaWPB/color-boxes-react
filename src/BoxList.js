import React, { useState } from 'react'
import Box from './Box'
import NewBoxForm from './NewBoxForm'
import {v4 as uuid} from 'uuid'

const BoxList = () => {
    const [boxes, setBoxes] = useState([])

    //function to run when new data is submited, adds a new box to the list.
    const addBox = box => {
        let newBox = { ...box, id: uuid() };
        setBoxes(boxes => [...boxes, newBox]);
      };

    return (
        <div className='BoxList'>
            <h1>Color Box Maker</h1>
            <div className='BoxList-form-div'>
                <h2>Create a box:</h2>
                <NewBoxForm addBox={addBox}/>
            </div>
            <div className='BoxList-boxes-div'>
                {boxes.map(box => <Box width={box.width} height={box.height} color={box.color} key={box.id}/>)}
            </div>
            
        </div>
    )
}

export default BoxList