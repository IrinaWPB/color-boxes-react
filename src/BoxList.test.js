import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import BoxList from './BoxList'

it('renders without crashing', () => {
    render(<BoxList />)
})

it('matches the snapshat', () => {
    const { asFragment } = render(<BoxList />)
    expect(asFragment()).toMatchSnapshot()
})

function addBox(boxList, height = "5", width = "5", color = "blue") {
    const heightInput = boxList.getByLabelText("Enter height:");
    const widthInput = boxList.getByLabelText("Enter width:");
    const backgroundInput = boxList.getByLabelText("Enter color:");
    fireEvent.change(backgroundInput, { target: { value: color } });
    fireEvent.change(widthInput, { target: { value: width } });
    fireEvent.change(heightInput, { target: { value: height } });
    const button = boxList.getByText("Add a box");
    fireEvent.click(button);
  }

it('shows created elment', () => {
    const boxes = render(<BoxList />)

    //no boxes
    expect(boxes.queryByTestId('blue-5-5')).not.toBeInTheDocument()

    //collect all the changes, clicking button to submit
    addBox(boxes)

    //box with id blue-5-5 shows on screen
    expect(boxes.getByTestId('blue-5-5')).toBeInTheDocument()
})