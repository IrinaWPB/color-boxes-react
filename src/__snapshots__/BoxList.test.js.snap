// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshat 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <h1>
      Color Box Maker
    </h1>
    <div
      class="BoxList-form-div"
    >
      <h2>
        Create a box:
      </h2>
      <form
        data-testid="form"
      >
        <label
          for="width"
        >
          Enter width:
        </label>
        <input
          id="width"
          name="width"
          placeholder="width"
          type="number"
          value=""
        />
        <label
          for="height"
        >
          Enter height:
        </label>
        <input
          id="height"
          name="height"
          placeholder="height"
          type="number"
          value=""
        />
        <label
          for="color"
        >
          Enter color:
        </label>
        <input
          id="color"
          name="color"
          placeholder="color"
          type="text"
          value=""
        />
        <br />
        <button>
          Add a box
        </button>
      </form>
    </div>
    <div
      class="BoxList-boxes-div"
    />
  </div>
</DocumentFragment>
`;
